api/callbacks.go

```diff
@@ -89,29 +102,43 @@ func buildAPI(api *GoAPI) C.GoApi {
 }
 
 //export cHumanAddress
-func cHumanAddress(ptr *C.api_t, canon C.Buffer, human C.Buffer) i32 {
+func cHumanAddress(ptr *C.api_t, canon C.Buffer, human *C.Buffer) (ret i32) {
+	// If the SDK panics, return -1 to inform the rust side something failed
+	defer func() { if recover() != nil { ret = -1 } }()
+	if human == nil {
+		// we received an invalid pointer
+		return -1
```

Comment:
You can also create FFI-safe enums for this. I'll indeed implement it here with simple named variants, and you can see how we have used complex enums for FFI in our codebase:
https://github.com/enigmampc/EnigmaBlockchain/blob/357e2d75befc41a5646e1b54e8e283db34dc2c09/cosmwasm/lib/enclave-ffi-types/src/types.rs#L34
(this type, all types it is built from, and the configuration files in the directory above this file)

Reply:
You don't explicitly set the numbers of those enums?
Which means if you change order then a compiled contract will no longer be in sync with the runtime. That is true if you change numbers as well, but people are usually more careful about changing explicit error codes, then reordering variables to be eg. alphabetically ordered.

Reply:
Oh, when defining the contract-runtime interface you definitely have to set explicit values, and you can do that with simple enums in rust.
But in this case, the interface is between two noninterchangeable (that's a long word) parts that are always shipped together, and their interface is not public to users of the module.

Reply:
I ended up setting explicit values anyways because i convert from i32 to the enum type, but keep in mind that the exact values aren't important for compatibility as long as the Rust and Go components are shipped together.